WIP Tech/overhaul ui

ndelangen opened the pull request:
# Refactor state-management en UI 

> Not ready for review yet

Fixes: https://github.com/storybooks/storybook/issues/3837 https://github.com/storybooks/storybook/issues/3854 https://github.com/storybooks/storybook/issues/3652 https://github.com/storybooks/storybook/issues/3702 https://github.com/storybooks/storybook/issues/3921 https://github.com/storybooks/storybook/issues/2313 https://github.com/storybooks/storybook/issues/3999 https://github.com/storybooks/storybook/issues/1708 https://github.com/storybooks/storybook/issues/3702 https://github.com/storybooks/storybook/issues/3921

Partially:
https://github.com/storybooks/storybook/issues/4139

### REFACTOR the state-management & core
- [x] REMOVE redux in preview, expand storyStore with it's function
- [x] REMOVE MANTRA in manager, use context for global state management
- [x] OPTIMIZE the way stories are stored in state
- [x] CLEANUP all addon names
- [ ] IMPROVE Graceful setStories, if partially fails show what was still succesfull
- [x] Log all errors to console when loading or running stories
- [x] Log important errors to notifications
- [x] FIX linting
- [x] CLEANUP storyStore api
- [x] CLEANUP containers / components from unused state
- [x] Add unit testing to Context components
- [x] FIX existing unit tests
- [x] CLEANUP setOptions everywhere to global addParameters
- [ ] INVESTIGATE if addon-storysource can use router directly over state method
- [x] router into a separate package
- [x] FIX when reloading page selected story should stay (this happens when there's a mismatch between hierarchySeparators)
- [ ] CLEANUP unify SET_CURRENT_STORY & SELECT_STORY
- [ ] FIX addons should clear when story changes and their data no longer applies
- [ ] ADD persistence of selected panel (not via url) upon refresh
- [x] ADD setting initial selected panel via url
- [x] single key shortcuts
- [x] FIX arrow shortcuts (should respond only to ALT + key)
- [x] configurable shortcuts ( @Jessica-Koch )

### HMR
- [x] HMR should work on component code
- [x] HMR should work on story code
- [x] HMR should work on imported markdown
- [x] FIX HMR - keep state of explorer
- [x] HMR should not change the url or selected story
- [x] HMR should work in ejected iframe

### New components
- [x] ADD SyntaxHighlighter component & use it everywhere where's there's code
- [x] USE SyntaxHighlighter in addon-notes
- [x] ADD Popout component for displaying menus
- [x] ADD Preview component
- [x] Copy button, when clicked copies text to clipboard, and use everywhere (extract from SyntaxHighlighter)

### IMPROVE the explorer
- [x] REMOVE treebeard, replace with a custom treeView component
- [x] REMOVE toNested (optimize story lookup)
- [x] ADD search in parameters https://github.com/storybooks/storybook/issues/4139
- [x] ADD support for rootSeperators
- [x] ADD auto-open of selection
- [ ] ADD auto select on click of group *contested as a feature, maybe behind option?*
- [x] ADD empty message when 0 stories loaded
- [x] ADD empty message when filtering results in 0	
- [ ] ADD explorer hover over items
- [x] ADD support for rootSeparators in explorer
- [ ] ADD highlighting of the search results
- [x] ADD fuzzy search capability

### Improve preview
- [x] ADD zoom function on preview component
- [x] ADD grid function on preview component
- [x] ADD eject iframe function to preview
- [x] ADD UI to toggle panels/fullscreen
- [x] ADD accessibility features (to demo new extensibility)
- [x] actually extract accessibility into an addon

### Update Notification
- [x] IMRPROVE styling (icon, padding)
- [x] ADD fetching of data for version notification
- [x] ADD storage of version info in localStorage
- [x] ADD api for adding and removing notifications

### Design
- [x] ADD tools/hotkeys button in nav
- [x] ADD UI to go fullscreen and toggle panels
- [x] CHANGE styles of main (preview+panel)
- [x] CHANGE styles of notifications
- [x] **ADD tabs to the preview toolbar & allow addons to render there**
- [x] CHANGE styles of explorer
- [x] CHANGE toolbar
- [x] ADD shortcut for toggle toolbar
- [ ] ADD shortcut for zoom
- [ ] ADD shortcut for toggle grid
- [x] ADD tools to TabsBar
- [x] ADD panel toggle to Panel Tabs
- [x] CHANGE icons
- [ ] IMPROVE theming
- [x] ADD theming to addon-notes

### Maybe, probably for another PR:
- [x] ADD ability to send postMessage to multiple iframes/ejected windows ( this would implement a follow-the-leader functionality)

### TODO:
- [ ] https://github.com/storybooks/storybook/issues/244
- [ ] https://github.com/storybooks/storybook/issues/866

### BUGS

 - [ ] Treeview does not reflect changes to storystore over HMR
 - [ ] Treeview does render stories in same order as they are defined
 - [ ] Treeview doesn't show highlight state as you control it via keyboard
 - [x] Custom shortcuts aren't reflected in menu
 - [.] No close button on about or shortcuts page
 - [ ] Styling on update page seems incomplete
 - [x] Actions addon seems extremely slow
 - [x] Jest addon seems to be broken
 - [x] Links addon seems to be broken

tmeasday commented:
@ndelangen -- is this reviewable? What would you like us to look at? The PR seems too big to look at

igor-dv commented:
@tmeasday, it's only a WIP. I think in the end there will be more :trollface:

alexandrebodin commented:
Yes just a start haha

igor-dv commented:
I've checked a bit what's going on here (😱). Can you at least separate the emotion upgrade out of the other changes?
Like: `master` -> `emotion-upgrade` -> `other-changes`

Hypnosphi commented:
Can you please avoid zebra-table style? It brings a lot more confusion than value

ndelangen commented:
> Can you please avoid zebra-table style? It brings a lot more confusion than value
it's themable!

Hypnosphi commented:
> It's something that was suggested by a designer..

Can you please @-mention the desiner here? I would really like to see the arguments

ndelangen commented on WORKLOG.md:
```diff
@@ -0,0 +1,35 @@
+@ndelangen
```
Remove before merge

ndelangen commented on addons/jest/src/register.js:
```diff
@@ -1,13 +1,15 @@
 import addons from '@storybook/addons';
 
-import PanelTitle from './components/PanelTitle';
+// import PanelTitle from './components/PanelTitle';
```
The Title as component idea is kinda broken, so I simplified it to just text.

Will visit this problem again later, and provide a proper solution to the "which addons are currently relevant to display in the tabs?"

ndelangen commented on examples/official-storybook/stories/core.stories.js:
```diff
@@ -16,13 +16,20 @@ storiesOf('Core|Parameters', module)
     }
   );
 
+storiesOf('Core|Decorator arguments', module)
```
This is actually a new feature!

Decorators can now call the `fn` with arguments, and those arguments are collected into 1 array, and injected into the context object.

Useful? yes | no | maybe so ?

tmeasday replied:
This is interesting @ndelangen -- I'd like to hear a bit more about it.

What does `context` look like before and after in this case? What is the use case?

I suspect people will want to at least be able to change all the `parameters` to a story from a decorator. Whether or not they should be able to do that (and if it'll do what they want depending on decorator order) is questionable.

ndelangen commented on lib/components/src/logo/logo.js:
```diff
@@ -0,0 +1,60 @@
+
+const silver = 'rgba(0,0,0,0.1)';
+
+const Logo = ({ colored, ...rest }) => (
```
This needs a tweak to make the S and bookmark white

ndelangen commented on lib/components/src/preview/preview.js:
```diff
@@ -0,0 +1,164 @@
+import React, { PureComponent, Component, Fragment } from 'react';
```
Maybe I should split this file into multiple files

ndelangen commented on lib/components/src/theme.js:
```diff
@@ -36,8 +42,16 @@ export const normal = {
     'linear-gradient(to bottom right, rgba(233, 233, 233, 0.6), rgba(255, 255, 255, 0.8))',
 
-  brand: {},
+  brand: {
```
So brand is the way you theme/ overload the logo, if you set this to `null` the text will be displayed

ndelangen commented on lib/components/src/treeview/treeview.js:
```diff
@@ -0,0 +1,301 @@
+import React, { PureComponent } from 'react';
```
I'm hoping to extract this component into it's own package and publish as `react-tree`

ndelangen commented on lib/core/src/client/preview/client_api.js:
```diff
@@ -9,7 +9,11 @@ import subscriptionsStore from './subscriptions_store';
-    (decorated, decorator) => context => decorator(() => decorated(context), context),
+    (decorated, decorator) => context =>
+      decorator(
+        (...args) => decorated({ ...context, arguments: [...context.arguments, ...args] }, ...args),
```
Here's the feature implemented to merge in arguments passed into the decorator `fn`

ndelangen commented on lib/core/src/client/preview/start.js:
```diff
@@ -62,84 +58,76 @@ function showException(exception) {
-    });
-    addons.setChannel(channel);
-    Object.assign(context, { channel });
+const isBrowser =
```
The iframe gets reloaded often when you switched between /settings and /components a lot.

I tried to make this process a bit more optimised.

ndelangen commented on lib/core/src/client/preview/start.js:
```diff
@@ -149,26 +137,23 @@ export default function start(render, { decorateStory } = {}) {
       forceRender,
     });
+
+    showMain();
```
Might be redundent

ndelangen commented on lib/core/src/client/preview/story_store.js:
```diff
@@ -45,26 +108,25 @@ export default class StoryStore extends EventEmitter {
   }
 
   getStoryKinds() {
```
UGH, all this still needs to be refactored TBH

ndelangen commented on lib/ui/README.md:
```diff
@@ -14,7 +14,6 @@ You can configure it by providing a provider API.
 -   [Hacking Guide](#hacking-guide)
```
80% of this 'guide' is outdated.. It's also pretty much never been useful to anyone I think.

I don't want to sink too much time into making this document correct..

ndelangen commented on lib/ui/src/app.js:
```diff
@@ -0,0 +1,53 @@
+import Panel from './containers/panel';
+
+const Reset = styled.div(({ theme }) => ({
+  fontFamily: theme.mainTextFace,
```
These css properties cascade down, meaning less stuff to do to get theming mostly right 99% of the places.

ndelangen commented on lib/ui/src/components/layout/desktop.js:
```diff
@@ -0,0 +1,136 @@
+import { localStorage } from 'global';
```
I'm thinking of overhauling still once more...

Currently enabling or disabling / moving panels has the undesired consequence that the iframe might reload.

IF...

I just place resize detectors inside the panels, and keep track of the desired sizes that way...
I can just absolutely position the true panels.

Hope this makes sense.

This would result in the panels never moving from the dom/reparenting.

ndelangen commented on lib/ui/src/components/nav/bar.js:
```diff
@@ -0,0 +1,46 @@
+import React from 'react';
```
Could make this a common component maybe?

ndelangen commented on lib/ui/src/init-key-handler.js:
```diff
@@ -0,0 +1,53 @@
+
+export default ({ store, uiStore }) => {
+  const handle = eventType => {
+    switch (eventType) {
```
Refactor into hash lookup

ndelangen commented on lib/ui/src/init-provider-api.js:
```diff
@@ -0,0 +1,65 @@
+  const api = {
+    onStory(cb) {
+      onStoryListeners.on('story', cb);
+      if (store.selectedKind && store.selectedStory) {
```
Need to refactor this

tmeasday commented:
cc @domyen

Hypnosphi commented:
@domyen which purpose do the zebra stripes have here?

igor-dv commented:
> ADD auto-open of selection

I don't like this feature, TBH. I don't want my stories to change when I am just opening the sub-tree... It should be at least optional

Another question:
![image](https://user-images.githubusercontent.com/7867954/45477299-eeb2f000-b749-11e8-9a83-7ba6cac73080.png)

I feel that this section, is likely to be extended, and there is no place for it in the stories panel. It will be nice to have it in the left side as the vertical tabs (like in the SBNext)

Also, I am missing highlighting of the search results.

domyen commented:
The images Norbert shared are WIP and not the final recommendation for the Storybook UI. I suspect lots of changes are underway, so I can't comment much on the specifics of what's currently on branch.

@Hypnosphi Stripes are intended to make the row distinctions clearer. They're a bit over the top right now, expect them to be toned down (a lot!).

@igor-dv Agree! :)

igor-dv commented:
@domyen, is there any sketch of the final UI that you can share?

domyen commented:
@igor-dv working on it right now! Give me a day or so.

ndelangen commented:
So here are the designs I'm going to implement on top of this branch:

<img width="1352" alt="screen shot 2018-09-13 at 17 25 03" src="https://user-images.githubusercontent.com/3070389/45498670-dad3b200-b77a-11e8-86db-053067f54fdb.png">
<img width="1347" alt="screen shot 2018-09-13 at 17 25 10" src="https://user-images.githubusercontent.com/3070389/45498668-dad3b200-b77a-11e8-9749-6902672f4b0c.png">
<img width="1345" alt="screen shot 2018-09-13 at 17 25 33" src="https://user-images.githubusercontent.com/3070389/45498666-dad3b200-b77a-11e8-8b3d-511dd848292a.png">
<img width="591" alt="screen shot 2018-09-13 at 17 25 42" src="https://user-images.githubusercontent.com/3070389/45498665-da3b1b80-b77a-11e8-9716-5c39a55e0d38.png">
<img width="902" alt="screen shot 2018-09-13 at 17 25 48" src="https://user-images.githubusercontent.com/3070389/45498664-da3b1b80-b77a-11e8-84c3-0f70f9a54e2b.png">
<img width="1339" alt="screen shot 2018-09-13 at 17 25 55" src="https://user-images.githubusercontent.com/3070389/45498663-da3b1b80-b77a-11e8-8496-758c038c8f53.png">

Hypnosphi commented:
@domyen But here each row is just one label. Does anyone have some trouble with distingushing them? The problem is, stripes bring the wrong impression like the odd rows are in some aspect different from even ones

Hypnosphi commented:
AFAIR @shilman used to have some objections agains file-like icons in tree view

shilman commented:
@Hypnosphi @domyen  I wasn't the biggest fan of the "heavy" file icons we had, but these ones are less visually intrusive and look pretty 👌 to me.

ndelangen committed:
ADD support for cyclic objects in channel && CLEANUP

Jessica-Koch committed:
capture keypress and render key on screen as well as fix the UI for keys

Jessica-Koch committed:
add ability to render keyboard shortcuts and set state to new state

alexandrebodin committed:
Remove mobx in favor of custom store with React context

ndelangen committed:
ADD decycling & support for functions, infinity, regex, instanced objects, undefined, null, symbol to channel

ndelangen committed:
IMPLEMENT API V2 && MIGRATE actions to use it && CLEANUP actions to make use of improved channel

ndelangen committed:
CLEANUP

ndelangen committed:
CLEANUP async addition of stories (still needs a proper api to work well)

ndelangen committed:
FIX panels && FIX a few missing key warnings

ndelangen committed:
Merge branch 'tech/overhaul-ui' of github.com:storybooks/storybook into tech/overhaul-ui

ndelangen committed:
REFACTOR stringify & parse & isJSON into a separate lib

ndelangen committed:
MOVE code to separate package (telejson)

Jessica-Koch committed:
#3984 add tests to key parsing functions

Jessica-Koch committed:
fix test

chinchiheather committed:
Fix issues with notes addon loading

chinchiheather committed:
Set selected state of tab according to route

alexandrebodin committed:
Split manager into more readable modules

Jessica-Koch committed:
#3984 keys now appear in the modal

Jessica-Koch committed:
Merge branch 'tech/overhaul-ui' of https://github.com/storybooks/storybook into tech/overhaul-ui

Jessica-Koch committed:
#3984 add tests and utiltity function to verify uniqueness of key shortcuts

ndelangen committed:
ADD toolbar hotkey

Jessica-Koch committed:
#3984 add default props to the keyboard , ability to clear input  and state value and a restore defaults function

Jessica-Koch committed:
#3984 consolidate buttons into one button with a case statemnet

Jessica-Koch committed:
#3984 add ability to restore defaults, determine who changed the shortcut, display error message if shortcut is a duplicate, undo actuion in modal

ndelangen committed:
Merge branch 'next' into tech/overhaul-ui

ndelangen committed:
FIX versions

ndelangen committed:
FIX syntaxhighlighting ( I think v10 is broken release )

ndelangen committed:
Merge branch 'next' into tech/overhaul-ui

ndelangen committed:
Merge branch 'next' into tech/overhaul-ui

ndelangen committed:
Merge branch 'next' into tech/overhaul-ui

Jessica-Koch committed:
Merge branch 'tech/overhaul-ui' of https://github.com/storybooks/storybook into tech/overhaul-ui

Jessica-Koch committed:
#3984 reduce number of keys and now input sets state, correctly throws error if shortcut cannot be saved, defaults restored on button click

ndelangen committed:
Merge branch 'next' into tech/overhaul-ui

# Conflicts:
#	lib/components/src/header/header.js
#	lib/components/src/icons/chevronRight.js
#	lib/components/src/layout/desktop.js
#	lib/components/src/layout/mobile.js
#	lib/components/src/navigation/MenuLink.js
#	lib/components/src/theme.js
#	lib/ui/src/modules/ui/components/stories_panel/__snapshots__/index.stories.storyshot
#	lib/ui/src/modules/ui/components/stories_panel/stories_tree/__snapshots__/index.stories.storyshot
#	lib/ui/src/modules/ui/components/stories_panel/stories_tree/tree_decorators.js
#	lib/ui/src/modules/ui/components/stories_panel/stories_tree/tree_header.js
#	lib/ui/src/modules/ui/components/stories_panel/text_filter.js

ndelangen committed:
FIX panel selection

ndelangen committed:
FIX treeview sorting (not sort stories, only groups and roots)

ndelangen committed:
Merge branch 'next' into tech/overhaul-ui

Jessica-Koch committed:
#3984 add tests to shortcuts file

Jessica-Koch committed:
#3984 save keyboard shortcuts to localstorage

Jessica-Koch committed:
#3984 nav menu changes when local storage items change

Jessica-Koch committed:
#3984 update tests and localstorage mock

Jessica-Koch committed:
#3984 fix error in restore defaults

ndelangen committed:
Merge branch 'next' into tech/overhaul-ui

# Conflicts:
#	jest.config.js

ndelangen committed:
Merge branch 'tech/overhaul-ui' of github.com:storybooks/storybook into tech/overhaul-ui

# Conflicts:
#	jest.config.js

ndelangen committed:
FIX missing keyboard shortcuts on first init

ndelangen committed:
FIX routing & layout

ndelangen committed:
FIX SafeTab and REFACTOR panelSelection

ndelangen committed:
CLEANUP

ndelangen committed:
FIX issue with knobs and possibly more addons

ndelangen committed:
Merge branch 'next' into tech/overhaul-ui

# Conflicts:
#	.babelrc
#	addons/actions/package.json
#	addons/backgrounds/package.json
#	addons/notes/package.json
#	lib/addons/package.json
#	lib/channel-websocket/package.json
#	lib/core/src/server/config/babel.dev.js
#	lib/ui/package.json

ndelangen committed:
FIX babel options for manager, unsure why it's suddenly broken here..

ndelangen committed:
FIX mobile viewport

ndelangen committed:
CHANGE build & dev-serve so both webpack compilation happen in the same process

CHANGE dev-serve so it watches manager

ndelangen committed:
Merge branch 'next' into tech/overhaul-ui

ndelangen committed:
REFACTOR dev-serve to output less noise

ndelangen committed:
CLEANUP

ndelangen committed:
Merge branch 'next' into tech/overhaul-ui

ndelangen committed:
FIX issues with legacy support of urls && REFACTOR stories api

ndelangen committed:
FIX backgrounds addon

ndelangen committed:
Merge branch 'next' into tech/overhaul-ui

# Conflicts:
#	addons/actions/package.json
#	addons/backgrounds/package.json
#	addons/notes/package.json
#	addons/storyshots/storyshots-core/package.json
#	lib/addons/package.json
#	lib/channel-websocket/package.json
#	lib/components/package.json
#	lib/ui/package.json

ndelangen committed:
Treeview fixes (very un-optimised)

Jessica-Koch committed:
#3984 add toggle toolbar state

ndelangen committed:
Merge branch 'next' into tech/overhaul-ui

# Conflicts:
#	CHANGELOG.md
#	addons/actions/package.json
#	addons/backgrounds/package.json
#	addons/notes/package.json
#	addons/storyshots/storyshots-core/package.json
#	addons/storyshots/storyshots-puppeteer/package.json
#	examples/cra-kitchen-sink/.storybook/config.js
#	examples/cra-kitchen-sink/src/stories/index.stories.js
#	lib/addons/package.json
#	lib/channel-websocket/package.json
#	lib/components/package.json
#	lib/components/src/addon_panel/__snapshots__/index.stories.storyshot
#	lib/components/src/header/__snapshots__/header.stories.storyshot
#	lib/components/src/layout/__snapshots__/index.stories.storyshot
#	lib/components/src/theme.js
#	lib/ui/package.json
#	lib/ui/src/modules/ui/components/stories_panel/__snapshots__/index.stories.storyshot
#	lib/ui/src/modules/ui/components/stories_panel/__snapshots__/text_filter.stories.storyshot
#	lib/ui/src/modules/ui/components/stories_panel/stories_tree/__snapshots__/index.stories.storyshot
#	lib/ui/src/modules/ui/configs/handle_routing.js
#	yarn.lock

Jessica-Koch committed:
Merge branch 'tech/overhaul-ui' of https://github.com/storybooks/storybook into tech/overhaul-ui

ndelangen committed:
Merge branch 'next' into tech/overhaul-ui

# Conflicts:
#	addons/actions/package.json
#	addons/backgrounds/package.json
#	addons/notes/package.json
#	addons/storyshots/storyshots-core/package.json
#	lib/addons/package.json
#	lib/channel-websocket/package.json
#	lib/components/package.json
#	lib/core/src/client/preview/story_store.js
#	lib/core/src/server/manager/manager-webpack.config.js
#	lib/ui/package.json
#	lib/ui/src/modules/api/configs/init_api.js
#	lib/ui/src/modules/ui/components/menu_item.stories.js
#	lib/ui/src/modules/ui/components/search_box.stories.js
#	lib/ui/src/modules/ui/components/shortcuts_help.stories.js
#	lib/ui/src/modules/ui/components/stories_panel/index.stories.js
#	lib/ui/src/modules/ui/components/stories_panel/stories_tree/index.stories.js
#	lib/ui/src/modules/ui/components/stories_panel/text_filter.stories.js

ndelangen committed:
CLEANUP babelrc

ndelangen committed:
CLEANUP manager.context & root component

ndelangen committed:
FIX lockfile

ndelangen committed:
ADD memoization in router for parsing componentParams

ndelangen committed:
WIP - CHANGE dependencies on react & react-dom to regular dependencies

in the manager we ensure there's a single version

ndelangen committed:
OPTIMIZE the ui

ndelangen committed:
OPTIMIZE treeview & explorer more

ndelangen committed:
OPTIMIZE more treeview & explorer

Jessica-Koch committed:
#3984 fix linting errors

Jessica-Koch committed:
#3984 fix linting errors

ndelangen committed:
REFACTOR addon-cssresources

lifeiscontent committed:
Upgrade emotion@10.0.2

lifeiscontent committed:
Update dependencies for knobs

lifeiscontent committed:
fix build

ndelangen committed:
CLEANUP && FIX issue if there are no cssresources

ndelangen committed:
CLEANUP

ndelangen committed:
REFACTOR viewports to use addParameters & only render tool

ndelangen committed:
REFACTOR addon-background

ndelangen committed:
Merge branch 'next' into tech/overhaul-ui

# Conflicts:
#	.babelrc.js
#	addons/a11y/package.json
#	addons/actions/package.json
#	addons/backgrounds/package.json
#	addons/cssresources/package.json
#	addons/events/package.json
#	addons/graphql/package.json
#	addons/info/package.json
#	addons/info/src/components/PropTable.js
#	addons/jest/package.json
#	addons/knobs/package.json
#	addons/links/package.json
#	addons/notes/package.json
#	addons/ondevice-backgrounds/package.json
#	addons/options/package.json
#	addons/storyshots/storyshots-core/package.json
#	addons/storyshots/storyshots-puppeteer/package.json
#	addons/storysource/package.json
#	app/angular/package.json
#	app/ember/package.json
#	app/html/package.json
#	app/marko/package.json
#	app/mithril/package.json
#	app/polymer/package.json
#	app/react/package.json
#	app/riot/package.json
#	app/svelte/package.json
#	app/vue/package.json
#	examples/official-storybook/stories/addon-a11y.stories.js
#	examples/official-storybook/stories/addon-actions.stories.js
#	lib/addons/package.json
#	lib/channel-postmessage/package.json
#	lib/channel-websocket/package.json
#	lib/components/package.json
#	lib/components/src/layout/mobile.js
#	lib/core/package.json
#	lib/core/src/server/common/babel.js
#	lib/ui/package.json
#	lib/ui/src/modules/ui/components/stories_panel/index.js
#	lib/ui/src/modules/ui/components/stories_panel/stories_tree/index.test.js
#	lib/ui/src/modules/ui/components/stories_panel/stories_tree/tree_decorators_utils.js
#	lib/ui/src/modules/ui/components/stories_panel/text_filter.js
#	yarn.lock

ndelangen committed:
FIX lockfile

ndelangen committed:
OPTIMIZE treeview state

ndelangen committed:
UPGRADE addon-background usage to use parameters everywhere

Jessica-Koch committed:
#3984 clear input when focusing, save on blur, and restores default value if nothing is in respective input, change error handling

Jessica-Koch committed:
#3984 change event handler

ndelangen committed:
Merge branch 'next' into tech/overhaul-ui

ndelangen committed:
Merge branch 'tech/overhaul-ui' of github.com:storybooks/storybook into tech/overhaul-ui

ndelangen committed:
Merge branch 'next' into tech/overhaul-ui

# Conflicts:
#	.eslintrc.js
#	addons/notes/package.json
#	addons/notes/src/index.ts
#	addons/notes/src/register.js
#	yarn.lock

Jessica-Koch committed:
#3984 persist keyboard shortcuts, update styles, add success handler

ndelangen committed:
REFACTOR addon-notes

ndelangen committed:
Merge branch 'tech/overhaul-ui' of github.com:storybooks/storybook into tech/overhaul-ui

ndelangen committed:
REFACTOR a11y addon && FIX issues with decorators reloading

ndelangen committed:
MODIFY option to use addParameters

ndelangen committed:
FIX a11y issue

kroeder committed:
migrated to ts

ndelangen committed:
REFACTOR state management

ndelangen committed:
Merge commit '4c5345d15cadb33e0277d399963c27bcee3b7623' into tech/overhaul-ui

ndelangen committed:
FIX option in story to hide panels

Jessica-Koch committed:
#3984 reduce code redundancy

ndelangen committed:
FIX iframe business, get the iframe once

ndelangen committed:
IMPROVE typescript types on addon-notes

ndelangen committed:
Merge branch 'tech/overhaul-ui' of github.com:storybooks/storybook into tech/overhaul-ui

ndelangen committed:
WIP

ndelangen committed:
REFACTOR addon-links

ndelangen committed:
REFACTOR addon-googleAnalytics && ADD error tracking and missing stories tracking

ndelangen committed:
CLEANUP

ndelangen committed:
Merge branch 'pr/lifeiscontent/4925' into tech/overhaul-ui

# Conflicts:
#	addons/a11y/package.json
#	addons/actions/package.json
#	addons/backgrounds/package.json
#	addons/cssresources/package.json
#	addons/events/package.json
#	addons/info/package.json
#	addons/info/src/__snapshots__/index.test.js.snap
#	addons/jest/package.json
#	addons/knobs/package.json
#	addons/notes/package.json
#	addons/viewport/package.json
#	app/react/package.json
#	examples/cra-kitchen-sink/package.json
#	examples/official-storybook/config.js
#	examples/official-storybook/package.json
#	lib/components/package.json
#	lib/components/src/layout/desktop.js
#	lib/components/src/layout/index.js
#	lib/core/package.json
#	lib/ui/package.json
#	lib/ui/src/modules/ui/components/search_box.js
#	lib/ui/src/modules/ui/components/shortcuts_help.js
#	lib/ui/src/modules/ui/components/stories_panel/__snapshots__/index.stories.storyshot
#	lib/ui/src/modules/ui/components/stories_panel/__snapshots__/text_filter.stories.storyshot
#	lib/ui/src/modules/ui/components/stories_panel/stories_tree/__snapshots__/index.stories.storyshot
#	lib/ui/src/modules/ui/components/stories_panel/stories_tree/tree_decorators.js
#	package.json
#	yarn.lock

ndelangen committed:
Merge branch 'next' into tech/overhaul-ui

# Conflicts:
#	addons/a11y/package.json
#	addons/actions/package.json
#	addons/backgrounds/package.json
#	addons/cssresources/package.json
#	addons/events/package.json
#	addons/google-analytics/package.json
#	addons/info/package.json
#	addons/jest/package.json
#	addons/knobs/package.json
#	addons/links/package.json
#	addons/notes/package.json
#	addons/storyshots/storyshots-core/package.json
#	addons/viewport/package.json
#	app/react/package.json
#	examples/cra-kitchen-sink/package.json
#	examples/official-storybook/package.json
#	examples/official-storybook/stories/addon-info.stories.js
#	lib/addons/package.json
#	lib/channel-websocket/package.json
#	lib/core/package.json
#	lib/ui/package.json

Jessica-Koch committed:
#3984 fix code so that it detcts modifier keys and keypresses

Jessica-Koch committed:
#3984 fix pushing duplicate keys

Jessica-Koch committed:
#3984 fix doubly adding key shortcuts to api

ndelangen committed:
Merge branch 'next' into tech/overhaul-ui

# Conflicts:
#	.eslintrc.js
#	lib/components/src/layout/__snapshots__/index.stories.storyshot
#	yarn.lock

Commit:
CMS-19849 refactor shortcuts to depend on arrays to respect ordering

Commit:
CMS-19849 add error handling

ndelangen committed:
Merge branch 'next' into tech/overhaul-ui

# Conflicts:
#	addons/a11y/package.json
#	addons/actions/package.json
#	addons/backgrounds/package.json
#	addons/cssresources/package.json
#	addons/events/package.json
#	addons/google-analytics/package.json
#	addons/info/package.json
#	addons/jest/package.json
#	addons/knobs/package.json
#	addons/links/package.json
#	addons/notes/package.json
#	addons/viewport/package.json
#	app/react/package.json
#	examples/cra-kitchen-sink/package.json
#	examples/official-storybook/package.json
#	lib/addons/package.json
#	lib/channel-websocket/package.json
#	lib/components/src/addon_panel/__snapshots__/index.stories.storyshot
#	lib/core/package.json
#	lib/ui/package.json
#	yarn.lock

ndelangen committed:
FIX some typescript issues

ndelangen committed:
FIX more typescript things

Jessica-Koch committed:
#3984 fix error when clicking modifiers, adjust styles and fix error message always being displayed

ndelangen committed:
FIX linting

ndelangen committed:
Merge branch 'next' into tech/overhaul-ui

ndelangen committed:
FIX more linting && FIX cyclic dependencies

ndelangen committed:
FIX snapshots

ndelangen committed:
MIGRATE storyshots to use new API with build-in context && update snapshots

ndelangen committed:
FIX tests

tmeasday commented on addons/storyshots/storyshots-core/src/api/index.js:
```diff
@@ -41,28 +35,51 @@ function testStorySnapshots(options = {}) {
-    testMethodParams,
-    snapshotSerializers,
-  });
+      if (parameters.storyshots !== false) {
```
This should be `parameters.storyshots && ! parameters.storyshots.disabled`.

We decided at some point to not set `parameters.X = false` to mean disabled, instead `parameters.X = { disabled: true }`. If I recall right it was you that advocated for that @ndelangen  ;)

ndelangen replied:
/shrug

Commit:
Not particularly successful attemps to get panel test passing

ndelangen committed:
FIX tests

Commit:
Remove deprecated URL APIs as they will no longer work

ndelangen committed:
FIX tests

ndelangen committed:
FIX tests

ndelangen committed:
Merge branch 'tech/overhaul-ui' of github.com:storybooks/storybook into tech/overhaul-ui

ndelangen committed:
Merge branch 'next' into tech/overhaul-ui

# Conflicts:
#	addons/a11y/package.json
#	addons/actions/package.json
#	addons/backgrounds/package.json
#	addons/cssresources/package.json
#	addons/events/package.json
#	addons/google-analytics/package.json
#	addons/info/package.json
#	addons/jest/package.json
#	addons/knobs/package.json
#	addons/links/package.json
#	addons/notes/package.json
#	addons/storyshots/storyshots-core/package.json
#	addons/viewport/package.json
#	app/react/package.json
#	docs/package.json
#	docs/src/pages/configurations/theming/index.md
#	examples/cra-kitchen-sink/package.json
#	examples/official-storybook/package.json
#	examples/preact-kitchen-sink/package.json
#	lib/addons/package.json
#	lib/channel-websocket/package.json
#	lib/components/package.json
#	lib/core/package.json
#	lib/ui/package.json
#	yarn.lock

tmeasday committed:
Merge 4848-new-example-format into tech/overhaul-ui

tmeasday committed:
FIX random emotion issue by updating version

tmeasday committed:
Revert "Remove deprecated URL APIs as they will no longer work"

This reverts commit 68f62c07c5ee2312470ef2502098586214e390d5.

cc @ndelangen -- this is still being used by knobs right now.

Should we keep the ability to get and set queryParams even though they don't appear in the URL for back compat?

If not, we need to fix knobs as we do this

tmeasday committed:
Fix up parameters examples

tmeasday committed:
Change .examples -> .stories

tmeasday committed:
Added a test for url module

@ndelangen -- note this is not passing. I'm not quite sure what's going on in this file. I am happy to fix it if you agree the tests are the correct behaviour.

tmeasday committed:
Added a test for queryParams ?

tmeasday committed:
Added tests for stories core api

Jessica-Koch committed:
#3984 fix tests

Jessica-Koch committed:
Merge branch 'tech/overhaul-ui' of github.com:storybooks/storybook into tech/overhaul-ui

tmeasday committed:
Added `selectStory` test

ndelangen committed:
Merge branch 'next' into tech/overhaul-ui

# Conflicts:
#	examples/cra-kitchen-sink/src/stories/__snapshots__/index.stories.storyshot
#	examples/official-storybook/stories/__snapshots__/addon-info.stories.storyshot
#	lib/core/package.json
#	yarn.lock

tmeasday committed:
remove getUrlData

tmeasday committed:
FIX up core/stories.js and tests

ndelangen committed:
Merge branch 'next' into tech/overhaul-ui

# Conflicts:
#	.eslintrc.js
#	addons/info/src/__snapshots__/index.test.js.snap
#	addons/viewport/src/manager/components/tests/viewportInfo.test.js
#	examples/cra-kitchen-sink/src/stories/__snapshots__/index.stories.storyshot
#	examples/official-storybook/stories/__snapshots__/addon-info.stories.storyshot
#	lib/core/package.json
#	package.json
#	yarn.lock

ndelangen committed:
FIX lockfile && FIX snapshots

ndelangen committed:
FIX storyshots

ndelangen committed:
FIX storyshots

tmeasday committed:
Use module.hot.dispose to do previousExports.

tmeasday committed:
Support legacy query params in url.js

tmeasday committed:
Added new id generation code

tmeasday committed:
Use id to route legacy urls

tmeasday committed:
Fix up stories to use new componentId

tmeasday committed:
Got url tests passing

tmeasday committed:
Added id to storystore

tmeasday committed:
Use new storyId format for story_store

tmeasday committed:
Change `componentId` => `storyId`

tmeasday committed:
Fix tests

ndelangen committed:
FIX tests

ndelangen committed:
FIX tests

ndelangen committed:
FIX tests

ndelangen committed:
FIX tests

ndelangen committed:
FIX tests

ndelangen committed:
FIX tests

ndelangen committed:
FIX tests

ndelangen committed:
FIX tests

ndelangen committed:
REMOVE tests, as it's no longer implemented here, instead in channel

ndelangen committed:
FIX tests

ndelangen committed:
REMOVE test that is essentially a duplicate of our story

ndelangen committed:
CLEANUP

ndelangen committed:
CHANGE to use yarn everywhere

ndelangen committed:
FIX first-child issue && ADD new Heading type

ndelangen committed:
FIX snapshots

ndelangen committed:
Merge branch 'next' into tech/overhaul-ui

# Conflicts:
#	addons/a11y/package.json
#	addons/actions/package.json
#	addons/backgrounds/package.json
#	addons/cssresources/package.json
#	addons/events/package.json
#	addons/google-analytics/package.json
#	addons/info/package.json
#	addons/info/src/__snapshots__/index.test.js.snap
#	addons/jest/package.json
#	addons/knobs/package.json
#	addons/links/package.json
#	addons/notes/package.json
#	addons/storyshots/storyshots-core/package.json
#	addons/storyshots/storyshots-core/src/frameworks/vue/renderTree.js
#	addons/viewport/package.json
#	app/angular/package.json
#	app/ember/package.json
#	app/html/package.json
#	app/marko/package.json
#	app/mithril/package.json
#	app/polymer/package.json
#	app/react/package.json
#	app/riot/package.json
#	app/svelte/package.json
#	app/vue/package.json
#	examples/cra-kitchen-sink/package.json
#	examples/official-storybook/package.json
#	examples/preact-kitchen-sink/package.json
#	examples/vue-kitchen-sink/src/stories/__snapshots__/addon-centered.stories.storyshot
#	examples/vue-kitchen-sink/src/stories/__snapshots__/custom-decorators.stories.storyshot
#	lib/addons/package.json
#	lib/channel-websocket/package.json
#	lib/components/package.json
#	lib/core/package.json
#	lib/ui/package.json
#	yarn.lock

ndelangen committed:
FIX linting

ndelangen committed:
FIX lockfile for docs

ndelangen committed:
REMOVE a ton of cruft from snapshots (reset & divs)

ndelangen committed:
REMOVE emotion from snapshots, as they become too big, and they provide little value

provided we're also using Chromatic

ndelangen committed:
FIX tests

Jessica-Koch committed:
#3984 create success and failure error messages

Jessica-Koch committed:
#3984 add submit actions to tab and enter keys as well as reimplement backspace functionality

ndelangen committed:
Merge branch 'tech/overhaul-ui' into tech/overhaul-ui-new-story-format

# Conflicts:
#	examples/official-storybook/config.js
#	examples/official-storybook/stories/core.stories.js

ndelangen committed:
Merge pull request #5058 from storybooks/tech/overhaul-ui-new-story-format

Use new story format in official storybook

ndelangen committed:
Merge branch '5.0.0' into tech/overhaul-ui

ndelangen committed:
FIX reset

ndelangen committed:
FIX color after reset fix

ndelangen committed:
REFACTOR to Typography component

Commit:
ADD parent field to story_store data

Commit:
Use parent field for treeview utils

ndelangen committed:
CLEANUP && FIX Tabs

ndelangen committed:
Merge branch '5.0.0' into tech/overhaul-ui

# Conflicts:
#	addons/a11y/package.json
#	addons/actions/package.json
#	addons/backgrounds/package.json
#	addons/backgrounds/src/__tests__/BackgroundPanel.js
#	addons/backgrounds/src/__tests__/Swatch.js
#	addons/cssresources/package.json
#	addons/events/package.json
#	addons/google-analytics/package.json
#	addons/info/package.json
#	addons/jest/package.json
#	addons/knobs/package.json
#	addons/knobs/src/components/__tests__/Panel.js
#	addons/links/package.json
#	addons/notes/package.json
#	addons/notes/src/index.ts
#	addons/notes/src/register.tsx
#	addons/notes/src/typings.d.ts
#	addons/storyshots/storyshots-core/package.json
#	addons/viewport/package.json
#	app/react/package.json
#	docs/package.json
#	docs/yarn.lock
#	examples/cra-kitchen-sink/package.json
#	examples/official-storybook/package.json
#	examples/preact-kitchen-sink/package.json
#	lib/addons/package.json
#	lib/addons/src/index.js
#	lib/channel-websocket/package.json
#	lib/components/package.json
#	lib/core/package.json
#	lib/ui/package.json
#	yarn.lock

ndelangen committed:
RENAME Link to link

ndelangen committed:
Merge branch '5.0.0' into tech/overhaul-ui

# Conflicts:
#	lib/channel-websocket/src/index.js
#	yarn.lock

ndelangen committed:
FIX gap in layout && FIX lib/addons

ndelangen committed:
FIX arrow keys shortcuts

ndelangen committed:
FIX filter disappear when 0 results

Jessica-Koch committed:
#3984 shortcuts are limited to 3 characters

Jessica-Koch committed:
Merge branch 'tech/overhaul-ui' of github.com:storybooks/storybook into tech/overhaul-ui

ndelangen committed:
FIX layout events handling

ndelangen committed:
Merge branch 'tech/overhaul-ui' of github.com:storybooks/storybook into tech/overhaul-ui

ndelangen committed:
REMOVE scrollbar in tabs list

ndelangen committed:
CLEANUP deprecated APIs

ndelangen committed:
RENAME selectedAddonPanel to selectedPanel

ndelangen committed:
FIX snapshots

ndelangen committed:
FIX parameter merging

ndelangen committed:
FIX skipping tests feature in storyshots

ndelangen committed:
IMPROVE warnings

ndelangen committed:
CHANGE core event names to uppercase && OPTIMISE app render

Commit:
Merge remote-tracking branch 'origin/tech/overhaul-ui' into tech/overhaul-ui-new-storyId

Jessica-Koch committed:
#3984 merge conflict

Jessica-Koch committed:
#3984 remove hidden input field

Commit:
Use correct prop in explorer.js

Commit:
Fix up key warning in addons panel

cc @ndelangen -- not sure if this was what you had in mind?

Commit:
Add notifications API

Commit:
More useful error

Commit:
Revert "Use correct prop in explorer.js"

This reverts commit ead61d8bab55a50bc5305166d5758d8ccfbe0342.

cc @ndelangen -- this fixes a props error but breaks the treeview. Not sure what was intended so I am backing it out.

Commit:
Clear the update notification when the about page renders

Commit:
Allow adding notifications

Jessica-Koch committed:
Merge branch 'tech/overhaul-ui' of github.com:storybooks/storybook into tech/overhaul-ui

Commit:
Add versions module and use it to set the versions notification

Commit:
Get version for about page from store

Jessica-Koch committed:
#3984 remove unnecessary console statements

libetl commented:
`INVESTIGATE if addon-storysource can use router directly over state method` What can I do to help ?

Jessica-Koch committed:
#3984 add more test
s

Jessica-Koch committed:
Merge branch 'tech/overhaul-ui' of github.com:storybooks/storybook into tech/overhaul-ui

Commit:
Add API for getting latest/current version

Commit:
Only highlight the menu button if there's a version update

Commit:
Merge remote-tracking branch 'origin/tech/overhaul-ui' into tech/overhaul-ui-new-storyId

Commit:
Final couple that I missed

Commit:
Merge pull request #5072 from storybooks/tech/overhaul-ui-new-storyId

Tech/overhaul ui new story

ndelangen committed:
FIX grid

ndelangen committed:
Merge tech/overhaul-ui

ndelangen committed:
Merge branch '5.0.0' into tech/overhaul-ui

# Conflicts:
#	addons/a11y/package.json
#	addons/actions/package.json
#	addons/backgrounds/package.json
#	addons/cssresources/package.json
#	addons/events/package.json
#	addons/google-analytics/package.json
#	addons/info/package.json
#	addons/jest/package.json
#	addons/knobs/package.json
#	addons/links/package.json
#	addons/notes/package.json
#	addons/storyshots/storyshots-core/package.json
#	addons/viewport/package.json
#	app/react/package.json
#	examples/cra-kitchen-sink/package.json
#	examples/official-storybook/package.json
#	examples/preact-kitchen-sink/package.json
#	lib/addons/package.json
#	lib/channel-websocket/package.json
#	lib/components/package.json
#	lib/core/package.json
#	lib/ui/package.json
#	yarn.lock

ndelangen committed:
UPGRADE dependencies

ndelangen committed:
FIX snapshots

ndelangen committed:
REFACTOR story_store

- CHANGE path to id
- MOVE code regarding to hierarchy into manager
- FIX HMR causing stories to be overridden (logging a warning)

ndelangen committed:
IMPROVE jest.init so it filters some errors (not failing a test over for it)

ndelangen committed:
Merge branch '5.0.0' into tech/overhaul-ui

# Conflicts:
#	addons/a11y/README.md
#	addons/a11y/package.json
#	addons/a11y/src/index.js
#	addons/actions/package.json
#	addons/actions/src/lib/decycle.js
#	addons/actions/src/lib/retrocycle.js
#	addons/actions/src/lib/reviver.js
#	addons/backgrounds/package.json
#	addons/cssresources/package.json
#	addons/events/package.json
#	addons/google-analytics/package.json
#	addons/info/package.json
#	addons/jest/package.json
#	addons/knobs/package.json
#	addons/links/package.json
#	addons/notes/package.json
#	addons/ondevice-backgrounds/package.json
#	addons/ondevice-knobs/package.json
#	addons/ondevice-notes/package.json
#	addons/options/package.json
#	addons/storyshots/storyshots-core/package.json
#	addons/storyshots/storyshots-puppeteer/package.json
#	addons/storysource/package.json
#	addons/viewport/package.json
#	app/react-native/package.json
#	app/react/package.json
#	examples/angular-cli/package.json
#	examples/cra-kitchen-sink/package.json
#	examples/mithril-kitchen-sink/package.json
#	examples/official-storybook/package.json
#	examples/preact-kitchen-sink/package.json
#	lib/addons/package.json
#	lib/channel-websocket/package.json
#	lib/components/package.json
#	lib/core/package.json
#	lib/ui/package.json
#	yarn.lock

ndelangen committed:
FIX story_store tests

tmeasday commented:
I did a little bit of QA on where we were at and added a list of bugs to the bottom on fhe PR summary above

Commit:
Use version endpoint to get version info

Commit:
Handle missing latest version on about pages

Commit:
Fix stories tests

ndelangen committed:
Merge branch '5.0.0' into tech/overhaul-ui

# Conflicts:
#	addons/a11y/package.json
#	addons/actions/package.json
#	addons/backgrounds/package.json
#	addons/cssresources/package.json
#	addons/events/package.json
#	addons/google-analytics/package.json
#	addons/info/package.json
#	addons/jest/package.json
#	addons/knobs/package.json
#	addons/knobs/src/components/__tests__/Panel.js
#	addons/links/package.json
#	addons/notes/package.json
#	addons/ondevice-backgrounds/package.json
#	addons/ondevice-knobs/package.json
#	addons/ondevice-notes/package.json
#	addons/options/package.json
#	addons/storyshots/storyshots-core/package.json
#	addons/storyshots/storyshots-puppeteer/package.json
#	addons/storysource/package.json
#	addons/viewport/package.json
#	app/react/package.json
#	examples/angular-cli/package.json
#	examples/cra-kitchen-sink/package.json
#	examples/mithril-kitchen-sink/package.json
#	examples/official-storybook/package.json
#	examples/preact-kitchen-sink/package.json
#	lib/addons/package.json
#	lib/channel-websocket/package.json
#	lib/client-logger/src/index.js
#	lib/components/package.json
#	lib/core-events/index.js
#	lib/core/package.json
#	lib/ui/package.json
#	lib/ui/src/modules/ui/components/stories_panel/stories_tree/__snapshots__/index.stories.storyshot
#	lib/ui/src/modules/ui/components/stories_panel/stories_tree/tree_decorators.js
#	lib/ui/src/modules/ui/components/stories_panel/stories_tree/tree_decorators_utils.js
#	lib/ui/src/modules/ui/components/stories_panel/stories_tree/tree_decorators_utils.test.js
#	yarn.lock

Commit:
Simplify treeview a little and handle input prop changes

ndelangen committed:
Merge pull request #5163 from storybooks/tmeasday/fix-treeview-reactivity

Simplify treeview a little and handle input prop changes

ndelangen committed:
Merge branch 'tech/overhaul-ui' of github.com:storybooks/storybook into tech/overhaul-ui

ndelangen committed:
Merge branch '5.0.0' into tech/overhaul-ui

# Conflicts:
#	examples/angular-cli/package.json
#	lib/channel-postmessage/src/index.ts
#	lib/core/package.json
#	package.json
#	yarn.lock
